spec/inputs/couchdb_changes_spec.rb

```diff
@@ -1 +1,507 @@
+  end
+
+  def deletedb
+    ftw = FTW::Agent.new
+    ftw.delete!("http://127.0.0.1:5984/db")
+  end
+
+  def populatedb
+    ftw = FTW::Agent.new
+    ftw.put!("http://127.0.0.1:5984/db/1", :body => '{"name":"Peter Parker"}')
+    ftw.put!("http://127.0.0.1:5984/db/2", :body => '{"name":"Mary Jane Watson"}')
+    ftw.put!("http://127.0.0.1:5984/db/3", :body => '{"name":"Captain America"}')
+    ftw.put!("http://127.0.0.1:5984/db/4", :body => '{"name":"J. Jonah Jameson"}')
+    ftw.put!("http://127.0.0.1:5984/db/5", :body => '{"name":"Otto Octavius"}')
+    ftw.put!("http://127.0.0.1:5984/db/6", :body => '{"name":"May Parker"}')
+    ftw.put!("http://127.0.0.1:5984/db/7", :body => '{"name":"Harry Osborne"}')
+    ftw.put!("http://127.0.0.1:5984/db/8", :body => '{"name":"Norman Osborne"}')
+    ftw.put!("http://127.0.0.1:5984/db/9", :body => '{"name":"Ben Parker"}')
+    ftw.put!("http://127.0.0.1:5984/db/10", :body => '{"name":"Stan Lee"}')
+  end
+
+  def updatedocs
+    ftw = FTW::Agent.new
+    data = ""
+    response = ftw.get!("http://127.0.0.1:5984/db/_changes?include_docs=true")
+    response.read_body { |chunk| data << chunk }
+    result = LogStash::Json.load(data)
+    result["results"].each do |doc|
+      upd = false
+      body = doc["doc"]
+      case doc["id"]
+      when "1"
+        body["Alter-ego"] = "Spider-man"
+        upd = true
+      when "5"
+        body["Alter-ego"] = "Doctor Octopus"
+        upd = true
+      when "8"
+        body["Alter-ego"] = "Green Goblin"
+        upd = true
+      end
+      if upd
+        ftw.put!("http://127.0.0.1:5984/db/#{doc["id"]}", :body => LogStash::Json.dump(body))
+      end
+    end
+  end
+
+  def deletedoc
+    ftw = FTW::Agent.new
+    data = ""
+    response = ftw.get!("http://127.0.0.1:5984/db/9")
+    response.read_body { |chunk| data << chunk }
+    doc = LogStash::Json.load(data)
+    ftw.delete!("http://127.0.0.1:5984/db/9?rev=#{doc["_rev"]}")
+  end
+
+  def createuser
+    ftw = FTW::Agent.new
+    ftw.put!("http://127.0.0.1:5984/_config/admins/logstash", :body => '"logstash"')
+  end
+
+  def deleteuser
+    user = "logstash"
+    pass = "logstash"
+    auth = "#{user}:#{pass}@"
+    ftw = FTW::Agent.new
+    ftw.delete!("http://#{auth}127.0.0.1:5984/_config/admins/logstash")
+  end
+
+  def deleteindex
+    ftw = FTW::Agent.new
+    ftw.delete!("http://127.0.0.1:9200/couchdb_test")
+  end
+
+  def buildup
+    # BEGIN: The following calls are a safety net in case of an aborted test
+    deleteuser
+    teardown
+    # END
+    createdb
+    populatedb
+  end
+
+  def teardown
+    deletedb
+    deleteindex
+    sequence = "/tmp/.couchdb_seq"
+    File.delete(sequence) if File.exist?(sequence)
+  end
+end
+    
+describe "inputs/couchdb_changes", :elasticsearch => true, :couchdb => true do
+  describe "Load couchdb documents", :elasticsearch => true, :couchdb => true do
+    include Helpers
+    sequence = "/tmp/.couchdb_seq"
+    index = "couchdb_test"
```

Comment:
use `let(:index) { "couchdb_test" }`. Prefer to randomize the index name if possible.